include/openthread/dataset.h

```diff
@@ -118,7 +122,11 @@ typedef otIp6NetworkPrefix otMeshLocalPrefix;
 OT_TOOL_PACKED_BEGIN
 struct otPskc
 {
-    uint8_t m8[OT_PSKC_MAX_SIZE]; ///< Byte values
+    union
```

Comment:
Thanks @hemanth-silabs for the new changes.

Regarding public APIs and definitions, I have some questions:

Here we define `otNetworkKey` and `otPskc` as a `union` of either literal key or PSA key ref.

These are then used in  a bunch of OT public APIs, like: `otThreadGetNetworkKey()` or `otThreadSetNetworkKey()` or in `otOperationalDataset`. We need to specify and add documentations in public headers/API on how the user should use the `union` type. 

For example when user sets the key `otThreadSetNetworkKey()`, 
- How would the user know whether to provide literal key or key ref?  
- Do we want to allow the user to be able to choose this? Or is it hard-coded and fixed by OT stack and its platform.  

Same situation when getting the key.

Reply:
@abtink Thank you for your suggestion. I agree, I will edit the documentation to highlight the union and the usage of the key literal and key references.

Right now the `otNetworkKey` and `otPskc` are being used in the following public APIs and modules.

- Operation dataset
- otThreadSetNetworkKey / otThreadSetPskc
- otThreadGetNetworkKey / otThreadGetPskc
- otThreadCopyNetworkKey / otThreadCopyPskc (New APIs added).

Right now the operational dataset has not changed, as the literal key is still being stored as part of the dataset. We will need to change this and remove the literal key from NVM but we decided to do it out of this PR. So I don't think anything needs to change in documentation for this.

Setters will still behave much the same way, with user setting the literal keys.

Getter function has changed significantly. I will update the setters and getters to indicate the proper usage.

Reply:
> Setters will still behave much the same way, with user setting the literal keys.

I think we should allow users to choose parameter type in setters (key literal or reference). But we can do it later, out of scope of this PR.